public/pages/Findings/containers/Findings/Findings.tsx

```diff
@@ -151,30 +218,74 @@ class Findings extends Component<FindingsProps, FindingsState> {
 
   onRefresh = async () => {
     await this.getNotificationChannels();
-    await this.getPlugins();
-    await this.getFindings();
+    if (this.state.selectedTabId === FindingTabId.DetectionRules) {
+      await this.getDetectionRulesFindings();
+    } else if (this.state.selectedTabId === FindingTabId.ThreatIntel) {
+      await this.getThreatIntelFindings();
+    }
     renderVisualization(this.generateVisualizationSpec(), 'findings-view');
   };
 
+  setStateForTab<T extends FindingTabId, F extends keyof FindingsState['findingStateByTabId'][T]>(
+    {
+      tabId,
+      field,
+      value,
+    }: { tabId: T; field: F; value: FindingsState['findingStateByTabId'][T][F] },
+    otherState?: Partial<Pick<FindingsState, keyof Omit<FindingsState, 'findingStateByTabId'>>>
+  ) {
+    this.setState({
+      ...(otherState as any),
+      findingStateByTabId: {
```

Comment:
same comment as above related to tabId

Reply:
Same reasoning, tabId makes more sense here to avoid overloading the term tab